includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -808,6 +810,14 @@ public function new_process_payment( WC_Order $order ) {
 		$service = wcpay_get_container()->get( PaymentProcessingService::class );
 		$state   = $service->process_payment( $order->get_id(), $manual_capture );
 
+		if ( $state instanceof PreviousPaidOrderDetectedState ) {
+			return [
+				'result'   => 'success',
+				'redirect' => $this->get_return_url( $order ), // TODO: This should be PaymentContext::get_previous_order_id(). But how to get it as we do expose PaymentContext here at the moment.
```

Comment:
@RadoslavGeorgiev - See 15dc9679b5eacf97c8e51f356eef1c50f3237ed7 - this is a tricky part. If we had a storage for PaymentContext, this would be easier.

Reply:
Handled here a8c478bc2e1d6c5a3f0df1dc0e5ca0e27eda65f3

Reply:
Resolving this now.